src/test/java/redis/clients/jedis/csc/ClientSideCacheFunctionalityTest.java

```diff
@@ -510,4 +510,30 @@ public void run() {
     }
   }
 
+  @Test
+  public void testNullValue() throws InterruptedException {
+    int MAX_SIZE = 20;
+    String nonExisting = "non-existing-key";
+    control.del(nonExisting);
+
+    TestCache cache = new TestCache(MAX_SIZE, new HashMap<>(), DefaultCacheable.INSTANCE);
+
+    try (JedisPooled jedis = new JedisPooled(hnp, clientConfig.get(), cache)) {
+      CacheStats stats = cache.getStats();
+
+      String val = jedis.get(nonExisting);
+      assertEquals(0, stats.getHitCount());
+      assertEquals(1, stats.getMissCount());
+
+      val = jedis.get(nonExisting);
+      assertEquals(1, stats.getHitCount());
+      assertEquals(1, stats.getMissCount());
+
+      control.set(nonExisting, "bar");
+      val = jedis.get(nonExisting);
+      assertEquals(1, stats.getHitCount());
+      assertEquals(2, stats.getMissCount());
```

Comment:
- assert(s) on cache size
- assert(s) on cache values; i.e. single entry and that entry contains a null value after both of first two GET.

Reply:
done